packages/logging-system/src/contexts/YLSProvider.tsx

```diff
@@ -32,6 +36,30 @@ export const YLSProvider = ({ children, baseURL }: YLSProviderProps) => {
     }
   };
 
+  useEffect(() => {
+    const handleVisibilityChange = () => {
+      if (document.visibilityState === 'visible') return;
+
+      const logList: LogType[] = JSON.parse(localStorage.getItem('yls-web') as string) || [];
+      const data: LogRequestList = {
+        logRequestList: logList,
+      };
+
+      SetLocalStorageClear();
+
+      void axiosInstance.put('/log/list', data, {
+        httpAgent: new http.Agent({ keepAlive: true }),
+        httpsAgent: new https.Agent({ keepAlive: true }),
+      });
+    };
+
+    document.addEventListener('visibilitychange', handleVisibilityChange);
+
+    return () => {
+      document.removeEventListener('visibilitychange', handleVisibilityChange);
+    };
+  }, []);
+
```

Comment:
`useEffect`를 사용해서 이벤트를 리스닝하게 되면 재랜더링시마다 이벤트를 재등록하게 되는데, `useRef`를 이용해서 등록하게 되면 그런 문제가 없을 듯 합니다.
```suggestion
  const documentRef = useRef(document);
  const logRequestListRef = useRef<LogRequestList>({ logRequestList: [] });
  function handleVisibilityChange() {
      if (documentRef.current.visibilityState === 'visible') return;

      const logRequestList: LogType[] = JSON.parse(localStorage.getItem('yls-web') as string) || [];
      logRequestListRef.current = {
        logRequestList,
      };
   }
  document.current.addEventListener('visibilitychange', handleVisibilityChange);
  useEffect(() => {
      SetLocalStorageClear();
      void axiosInstance.put('/log/list', logRequestListRef.current, {
        httpAgent: new http.Agent({ keepAlive: true }),
        httpsAgent: new https.Agent({ keepAlive: true }),
      });
  }, []);

```

Reply:
useRef로 document를 참조해도 결국 YLSProvider 컴포넌트가 리렌더링 되면 ref가 참조하는 메서드를 다시 실행하니까 이벤트 리스너가 재등록되지 않나요?

Reply:
[DOM Standard](https://dom.spec.whatwg.org/#ref-for-dom-eventtarget-addeventlistener%E2%91%A2)에 보시면 아래와 같은 내용이 있습니다.

> The event listener is appended to target’s event listener list and is not appended if it has the same type, callback, and capture.

말씀하신 대로 매번 addEventListener가 실행되겠지만, useCallback을 empty deps로 사용해서 handleVisibilityChange 함수를 넘겨주면 중복해서 리스너가 bind되지는 않겠네요.

Reply:
하나 또 배워갑니다...